tests/test_application.py

```diff
@@ -53,7 +53,18 @@
     filters,
 )
 from telegram.warnings import PTBUserWarning
-from tests.conftest import PROJECT_ROOT_PATH, call_after, make_message_update, send_webhook_message
+from tests.conftest import (
+    PROJECT_ROOT_PATH,
+    call_after,
+    make_bot,
+    make_message_update,
+    send_webhook_message,
+)
+
+
+@pytest.fixture(scope="function")
+def test_bot(bot_info):  # was needed since we get HTTPXRequest not initialized after app tests
+    return make_bot(bot_info)
```

Comment:
1. comment -> docstring
2. I don't understand the comment and don't recall what the problem was 😄 Can you elaborate?
3. `test_bot` is not a very speaking name. I'd like to improve that, but that depends on the explanation for 2 …

Reply:
2. For whatever reason, if the session `bot` is used (from conftest), these app tests would work fine, but the test files executed after `test_application.py` would fail saying that `HTTPXRequest` was not initialized. So I just resorted to making a new bot. 

P.S: I think I figured out why that happened, I think it's cause of:
```
bot = ApplicationBuilder().bot(bot).build()
async with app:
```
done in a few tests.

3. Yes, don't know why I named it starting with `test_` 😞

Reply:
ah okay, that makes sense, the problem is that `bot.shutdown` was called. How about `disposable_bot` or `one_time_bot` or `function_scope_bot` then to signal that we get a new bot every time we use that fixture?

Reply:
sure